deeplearning4j-core/src/test/java/org/deeplearning4j/gradientcheck/CNN1DGradientCheckTest.java

```diff
@@ -32,6 +33,83 @@
+            for (SubsamplingLayer.PoolingType poolingType : poolingTypes) {
+                for (int minibatchSize : minibatchSizes) {
+                    for (int kernel : kernels) {
+                        INDArray input = Nd4j.rand(new int[] {minibatchSize, convNIn, length});
+                        INDArray labels = Nd4j.zeros(minibatchSize, finalNOut, croppedLength);
+                        for (int i = 0; i < minibatchSize; i++) {
+                            for (int j = 0; j < croppedLength; j++) {
+                                labels.putScalar(new int[] {i, i % finalNOut, j}, 1.0);
+                            }
+                        }
+
+                        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
+                                .updater(new NoOp()).weightInit(WeightInit.DISTRIBUTION)
+                                .dist(new NormalDistribution(0, 1)).convolutionMode(ConvolutionMode.Same).list()
+                                .layer(new Convolution1DLayer.Builder().activation(afn).kernelSize(kernel)
+                                        .stride(stride).padding(padding).nIn(convNIn).nOut(convNOut1)
+                                        .build())
+                                .layer(new Cropping1D.Builder(cropping).build())
+                                .layer(new Convolution1DLayer.Builder().activation(afn).kernelSize(kernel)
+                                        .stride(stride).padding(padding).nIn(convNOut1).nOut(convNOut2)
+                                        .build())
+                                .layer(new RnnOutputLayer.Builder(LossFunctions.LossFunction.MCXENT)
+                                        .activation(Activation.SOFTMAX).nOut(finalNOut).build())
+                                .setInputType(InputType.recurrent(convNIn)).build();
+
+                        String json = conf.toJson();
+                        MultiLayerConfiguration c2 = MultiLayerConfiguration.fromJson(json);
+                        assertEquals(conf, c2);
+
+                        MultiLayerNetwork net = new MultiLayerNetwork(conf);
+                        net.init();
+
+                        String msg = "PoolingType=" + poolingType + ", minibatch=" + minibatchSize + ", activationFn="
+                                + afn + ", kernel = " + kernel;
+
+                        if (PRINT_RESULTS) {
+                            System.out.println(msg);
+                            for (int j = 0; j < net.getnLayers(); j++)
+                                System.out.println("Layer " + j + " # params: " + net.getLayer(j).numParams());
+                        }
+
+                        boolean gradOK = GradientCheckUtil.checkGradients(net, DEFAULT_EPS, DEFAULT_MAX_REL_ERROR,
+                                DEFAULT_MIN_ABS_ERROR, PRINT_RESULTS, RETURN_ON_FIRST_FAILURE, input, labels);
+
+                        assertTrue(msg, gradOK);
```

Comment:
These days: I'm tending to add a ```TestUtils.testModelSerialization(net);```  at the end of gradient checks as an extra check (if only to ensure a bunch of different configurations are tested *somewhere* for every layer)